### PR TITLE
[#175573274] Allow AI tracking on service's visibility

### PR DESCRIPTION
**Pull request by AleDore**
This PR introduces Application Insights tracking on Create/Update Service APIs with a specific focus to Service visibility, due to security's monitoring purpose.

### CODE REVIEW

File: UpdateService/handler.ts [hunk 1]
```diff
@@ -148,9 +150,22 @@ const updateServiceTask = (
...
 ): IUpdateServiceHandler {
   return (_, apiAuth, ___, userAttributes, serviceId, servicePayload) => {
+    const trackUpdateServiceVisibility = (
+      isVisible: boolean,
+      svcId: ServiceId
+    ): void =>
+      telemetryClient.trackEvent({
+        name: "api.services.update",
+        properties: {
+          isVisible: isVisible ? "false" : String(isVisible),
```
Comment (balanza) [inline UpdateService/handler.ts:164]: I'm sure there's meaning on that, but I cannot really get it

File: UpdateService/handler.ts [hunk 2]
```diff
@@ -173,8 +188,11 @@ export function UpdateServiceHandler(
                 userInfo.token_name
               )
             )
-            .chain(service =>
-              getSubscriptionKeysTask(
+            .chain(service => {
```
Comment (balanza) [inline UpdateService/handler.ts:191]: Can we flat this chain-pipeline? In theory, the following
```ts
.chain(
  foo1.chain(
    foo2.chain(
       foo3
    )
  )
)
```

It's equal to 
```ts
.chain( foo1 )
.chain( foo2 )
.chain( foo3 )

File: UpdateService/handler.ts (hunk 1 above)
Comment (AleDore) [reply UpdateService/handler.ts:164]: Nope you're right 🤣

File: UpdateService/handler.ts [hunk 3]
```diff
@@ -161,7 +161,7 @@ export function UpdateServiceHandler(
       telemetryClient.trackEvent({
         name: "api.services.update",
         properties: {
-          isVisible: isVisible ? "false" : String(isVisible),
+          isVisible: isVisible ? String(isVisible) : "false",
```
Comment (balanza) [inline UpdateService/handler.ts:164]: isn't this enough
```suggestion
          isVisible: String(isVisible)
```
or even

```suggestion
          isVisible
```

File: UpdateService/handler.ts (hunk 2 above)
Comment (AleDore) [reply UpdateService/handler.ts:191]: In order to flat chain-pipeline we have to build a struct that contains a subset of right values for each chain, due to usage of this values in the last chain ex:
```ts
.chain(() =>
  foo1.chain(bar1 =>
    foo2.chain(bar2 =>
       foo3(bar1, bar2)
    )
  )
)
```

File: UpdateService/handler.ts (hunk 3 above)
Comment (AleDore) [reply UpdateService/handler.ts]: Fixed in [821fa9e](https://github.com/pagopa/io-functions-services/pull/87/commits/821fa9e7bdef60a9b32e956dac12af7d1eacd21a)

File: UpdateService/handler.ts [hunk 4]
```diff
@@ -148,9 +150,22 @@ const updateServiceTask = (
  * Handles requests for updating a service by given serviceId and a Service Payload.
  */
 export function UpdateServiceHandler(
+  telemetryClient: ReturnType<typeof initAppInsights>,
   apiClient: APIClient
 ): IUpdateServiceHandler {
   return (_, apiAuth, ___, userAttributes, serviceId, servicePayload) => {
+    const trackUpdateServiceVisibility = (
```
Comment (gunzip) [inline UpdateService/handler.ts:157]: there's no need to wrap `telemetryClient.trackEvent` into a closure, just use it inline

**Review by gunzip — CHANGES_REQUESTED** [review]

File: UpdateService/handler.ts (hunk 4 above)
Comment (AleDore) [reply UpdateService/handler.ts]: Fixed in [6aae3a7](https://github.com/pagopa/io-functions-services/pull/87/commits/6aae3a75463140ef02f6ed99c95de6ba45e9d011)

**Review by gunzip — APPROVED** [review]

**Review by balanza — APPROVED** [review]
lgtm

**Pull request merged**